libr/include/r_types.h

```diff
@@ -227,8 +227,9 @@ typedef void (*PrintfCallback)(const char *str, ...);
 
 /* TODO: Move outside */
 #define _perror(str,file,line) \
-  { char buf[128];snprintf(buf,sizeof(buf),"%s:%d %s",file,line,str);perror(buf); }
+  { char buf[128];snprintf(buf,sizeof(buf),"%s:%d %s",file,line,str);r_sys_perror_str(buf); }
 #define perror(x) _perror(x,__FILE__,__LINE__)
+#define r_sys_perror(x) _perror(x,__FILE__,__LINE__)
```

Comment:
What about adding also `__func__` here? Would be helpful to understand without looking into the file and search by the line number?
Like `__FILE__, __func__, __LINE__`?

Don't forget to add also
```
#ifdef _MSC_VER
#define __func__ __FUNCTION__
#endif
```
for the sake of compiling under MSVC.